quipucords/utils/deepget.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2022  Red Hat, Inc.
+
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+
+"""Module for deepget function."""
+
+
+def _get_item(data, key):
+    try:
+        return data[key]
+    except TypeError:
+        try:
+            key = int(key)
+        except ValueError:
+            return None
+        return _get_item(data, key)
+    except (IndexError, KeyError):
+        return None
+
+
+def _get_nested_item(data, *keys):
+    if len(keys) == 1:
+        current_key = keys[0]
+        return _get_item(data, current_key)
+
+    current_key, *next_keys = keys
+    item = _get_item(data, current_key)
+    if getattr(item, "__getitem__", None) and not isinstance(item, str):
```

Comment:
it will return a function if `item` has `__getitem__` implemented.

objects that allow accessing like this `item[some_element]` have this implement.

going back to `str`, they also have a `__getitem__` method -> that's the reason I had to put this stop condition.

Reply:
Cool, makes sense. I'm slowly learning these idioms. Thanks for the explanation!